live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializerTest.java

```diff
@@ -185,6 +190,43 @@ public void shouldOnlySendUpdatedContentMetadata()
         assertJsonEquals(expectedJson, actualJson);
     }
 
+    @Test
+    public void shouldReturnCorrectTypeForVariousValues()
+    {
+        assertAll(
+                () -> assertEquals("boolean", serializer.selectTypeByValue(true)),
+                () -> assertEquals("integer", serializer.selectTypeByValue(123)),
+                () -> assertEquals("float", serializer.selectTypeByValue(123.45f)),
+                () -> assertEquals("string", serializer.selectTypeByValue("testTypeValue")));
+    }
+
+    @Test
+    public void shouldThrowExceptionForEmptyCollection()
+    {
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+            serializer.selectTypeByValue(Collections.emptyList());
+        });
+        assertEquals("Empty collections should not be passed to selectTypeByValue.", exception.getMessage());
+    }
+
+    @Test
+    public void shouldReturnStringForMixedTypeCollection()
+    {
+        assertEquals("string", serializer.selectTypeByValue(Arrays.asList(1, "test", 3.14)));
```

Comment:
I don't see how this works. The code looks like it should select an arbitrary entry from the list and use that:
https://github.com/Alfresco/hxinsight-connector/pull/888/files#diff-d1784b0b5262c967b22e4bfffe85820bb1d1bc0241e354dcd92f48c4776b2250R187

I hadn't thought about mixed type collections (I'm not sure whether it's possible in Alfresco with `d:any`), but regardless this is a neat idea for a test.